File: MathMatchupBackend/src/main/java/com/brendanharan/mathmatchupbackend/players/analytics/PointsAndAnalytics.java
/*
 * Copyright (c) 2025 by Brendan Haran, All Rights Reserved.
 * Use of this file or any of its contents is strictly prohibited without prior written permission from Brendan Haran.
 * Current File (PointsAndAnalytics.java) Last Modified on 2/9/25, 2:50 PM
 *
 */

package com.brendanharan.mathmatchupbackend.players.analytics;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class PointsAndAnalytics {
    private int points;
    private int correctAnswers;
    private int incorrectAnswers;
    private int totalQuestions;
    private double averageTime;
    private double accuracy;
}

File: math_matchup/lib/src/features/question_page/data/points_and_analytics.dart
/*
 * Copyright (c) 2025 by Brendan Haran, All Rights Reserved.
 * Use of this file or any of its contents is strictly prohibited without prior written permission from Brendan Haran.
 * Current File (points_and_analytics.dart) Last Modified on 2/9/25, 1:41 PM
 *
 */
class PointsAndAnalytics {
  int points;
  int correctAnswers;
  int incorrectAnswers;
  int totalQuestions;
  double averageTime;
  double accuracy;

  PointsAndAnalytics({
    required this.points,
    required this.correctAnswers,
    required this.incorrectAnswers,
    required this.totalQuestions,
    required this.averageTime,
    required this.accuracy,
  });
}
